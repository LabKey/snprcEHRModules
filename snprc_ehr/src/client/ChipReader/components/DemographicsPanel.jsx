import React from 'react'
import Select from 'react-select'
import moment from 'moment'
import WrappedDatePicker from '../../Shared/components/WrappedDatePicker'
import InfoPanel from '../../Shared/components/InfoPanel'

export default class DemographicsPanel extends React.Component {

    state = {
        errorMessage: undefined
    }

    componentDidMount = () => {
      console.log('DemographicsPanel mounted')
    }

    render() {
        return (
          <>
            <div className="wizard-panel__rows">
              <div className="wizard-panel__row">
                <div className="wizard-panel__col">
                  <p>Demographic Data Here</p>

                </div>
              </div>

            </div>
            <InfoPanel
              errorMessages={ this.state.errorMessage
                        && [{ propTest: true, colName: this.state.errorMessage }] }

            />
          </>
        )
    }
}
