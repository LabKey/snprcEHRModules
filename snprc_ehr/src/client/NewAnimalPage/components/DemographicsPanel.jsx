import React from 'react';
import Select from 'react-select';
import WrappedDatePicker from '../../utils/components/WrappedDatePicker';

export default class DemographicsPanel extends React.Component {
    state = {
        selectedDate: this.props.newAnimalData.birthDate || new Date()
    }

    handleDamChange = option => {
        this.props.handleDataChange('dam', option);
    }
    handleSireChange = option => {
        this.props.handleDataChange('sire', option);
    }
    handleGenderChange = option => {
        this.props.handleDataChange('gender', option);
    }
    handleBirthDateChange = (date) => {
        this.handleDateSelect(date);
        this.props.handleDataChange('birthDate', date);
    }

    handleDateSelect = date => {
        this.setState({
            selectedDate: date
        });
    };


    render() {
        let {gender, dam, sire} = this.props.newAnimalData;
        return (
            <div className="wizard-panel__rows">
                <div className="wizard-panel__row" >
                    <div className="wizard-panel__col">
                        <WrappedDatePicker
                            label="Birthdate"
                            todayButton="Today"
                            showTimeSelect
                            timeFormat="p"
                            timeIntervals={30}
                            dateFormat="Pp"
                            maxDate={new Date()}
                            selected={this.state.selectedDate} 
                            onSelect={this.handleDateSelect}
                            onChange={this.handleBirthDateChange}
                            disabled={this.props.disabled}
                        />
                    </div>
                </div>
                <div className="wizard-panel__row" >
                    <div className="wizard-panel__col">
                        <label className="field-label" >Gender</label>
                        <Select
                            defaultValue={gender}
                            className="shared-dropdown"
                            classNamePrefix="shared-select"
                            options={[{ value: "F", label: "Female" },
                            { value: "M", label: "Male" },
                            { value: "U", label: "Unknown" }]}
                            onChange={this.handleGenderChange}
                            placeholder="Select Gender"
                            isDisabled={this.props.disabled}
                            id="gender-select"
                        />
                    </div>
                </div>

                <div className="wizard-panel__row" >
                    <div className="wizard-panel__col">
                        <label className="field-label" >Dam</label>
                        <Select
                            defaultValue={dam}
                            className="shared-dropdown"
                            classNamePrefix="shared-select"
                            options={this.props.potentialDamList}
                            onChange={this.handleDamChange}
                            placeholder="Select Dam"
                            isDisabled={this.props.disabled}
                            id="dam-select"
                        />
                    </div>
                </div>

                <div className="wizard-panel__row" >
                    <div className="wizard-panel__col">
                        <label className="field-label" >Owner</label>
                        <Select
                            defaultValue={sire}
                            className="shared-dropdown"
                            classNamePrefix="shared-select"
                            options={this.props.potentialSireList}
                            onChange={this.handleSireChange}
                            placeholder="Select Sire"
                            isDisabled={this.props.disabled}
                            id="sire-select"
                        />
                    </div>
                </div>
            </div>

        )
    }
}
