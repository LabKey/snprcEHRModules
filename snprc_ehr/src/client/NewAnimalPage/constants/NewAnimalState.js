import moment from 'moment';

const NewAnimalState = () => {

    return  ({
        currentPanel: 1,
        selectedOption: undefined,
        newAnimalData: {
            selectedOption: undefined,
            id: undefined,
            birthDate: { date: moment() },
            bdStatus: undefined,
            acquisitionType: undefined,
            acqDate: { date: moment() },
            gender: undefined,
            sire: undefined,
            dam: undefined,
            species: undefined,
            colony: undefined,
            animalAccount: undefined,
            ownerInstitution: undefined,
            responsibleInstitution: undefined,
            room: undefined,
            cage: { value: undefined },
            diet: undefined,
            pedigree: undefined,
            iacuc: undefined
        },
        summaryData: [],
        speciesList: [],
        acquisitionTypeList: [],
        potentialDamList: [],
        potentialSireList: [],
        locationList: [],
        accountList: [],
        colonyList: [],
        dietList: [],
        institutionList: [],
        iacucList: [],
        pedigreeList: [],
        bdStatusList: [],
        isLoading: true,
        hasError: false,
        preventNext: true,
        saveOk: false,
        showSaveModal: false,
        showCancelModal: false,
        showSpeciesChangeModal: false,
        errorMessage: undefined
    });
}

export default NewAnimalState;
