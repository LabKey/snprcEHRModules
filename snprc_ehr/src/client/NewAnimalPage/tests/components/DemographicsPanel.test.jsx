import React from 'react'
import { shallow } from 'enzyme'
import DemographicsPanel from '../../components/DemographicsPanel'
import NewAnimalState from '../../constants/NewAnimalState'
import lists from '../fixtures/lists'

jest.mock('moment', () => {
    return () => jest.requireActual('moment')('2023-01-01T00:00:00.000Z');
});

test('Should render the DemographicsPanel', () => {
    const { newAnimalData } = new NewAnimalState()

    const wrapper = shallow(
      <DemographicsPanel
        bdStatusList={ lists.bdStatusList }
        disabled={ false }
        handleDataChange={ () => { return false } }
        newAnimalData={ newAnimalData }
        potentialDamList={ lists.potentialDamList }
        potentialSireList={ lists.potentialSireList }
        preventNext={ () => { return false } }
        reloadDamsAndSires={ () => { return false } }
      />
    )
    expect(wrapper).toMatchSnapshot()
})
