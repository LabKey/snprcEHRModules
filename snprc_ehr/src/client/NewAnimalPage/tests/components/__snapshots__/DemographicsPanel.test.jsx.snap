// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the DemographicsPanel 1`] = `
<Fragment>
  <div
    className="wizard-panel__rows"
  >
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <WrappedDatePicker
          dateFormat="Pp"
          disabled={false}
          label="Birthdate"
          maxDate={2023-01-01T00:00:00.000Z}
          onChange={[Function]}
          onSelect={[Function]}
          selected={2023-01-01T00:00:00.000Z}
          showTimeSelect={true}
          timeFormat="p"
          timeIntervals={30}
          todayButton="Today"
        />
      </div>
    </div>
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <label
          className="field-label"
        >
          Birthdate Status
        </label>
        <ForwardRef
          autoFocus={true}
          className="shared-dropdown"
          classNamePrefix="shared-select"
          id="bdStatus-select"
          isClearable={true}
          isDisabled={false}
          isLoading={false}
          onChange={[Function]}
          options={
            [
              {
                "id": 0,
                "label": "Day/Month/Year OK",
                "value": 0,
              },
              {
                "id": 1,
                "label": "Month/Year OK",
                "value": 1,
              },
            ]
          }
          placeholder="Select Birthdate Status"
        />
      </div>
    </div>
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <label
          className="field-label"
        >
          Gender
        </label>
        <ForwardRef
          className="shared-dropdown"
          classNamePrefix="shared-select"
          id="gender-select"
          isClearable={true}
          isDisabled={false}
          onChange={[Function]}
          options={
            [
              {
                "label": "Female",
                "value": "F",
              },
              {
                "label": "Male",
                "value": "M",
              },
              {
                "label": "Unknown",
                "value": "U",
              },
            ]
          }
          placeholder="Select Gender"
        />
      </div>
    </div>
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <label
          className="field-label"
        >
          Dam
        </label>
        <ForwardRef
          className="shared-dropdown"
          classNamePrefix="shared-select"
          id="dam-select"
          isClearable={true}
          isDisabled={false}
          isLoading={false}
          onChange={[Function]}
          options={
            [
              {
                "Age": 24.7,
                "ArcSpeciesCode": "PC",
                "id": 0,
                "label": "12924",
                "value": "12924",
              },
              {
                "Age": 21.8,
                "ArcSpeciesCode": "PC",
                "id": 1,
                "label": "12925",
                "value": "12925",
              },
            ]
          }
          placeholder="Select Dam"
        />
      </div>
    </div>
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <label
          className="field-label"
        >
          Sire
        </label>
        <ForwardRef
          className="shared-dropdown"
          classNamePrefix="shared-select"
          id="sire-select"
          isClearable={true}
          isDisabled={false}
          isLoading={false}
          onChange={[Function]}
          options={
            [
              {
                "Age": 22.8,
                "ArcSpeciesCode": "PC",
                "id": 0,
                "label": "14022",
                "value": "14022",
              },
              {
                "Age": 19.8,
                "ArcSpeciesCode": "PC",
                "id": 1,
                "label": "14023",
                "value": "14023",
              },
            ]
          }
          placeholder="Select Sire"
        />
      </div>
    </div>
  </div>
  <InfoPanel
    messages={
      [
        {
          "colName": "Birthdate Status",
          "propTest": true,
        },
        {
          "colName": "Gender",
          "propTest": true,
        },
      ]
    }
  />
</Fragment>
`;
